Messaging/Plugin.Messaging.Android/SmsTask.cs

```diff
@@ -38,16 +61,57 @@ public void SendSms(string recipient = null, string message = null)
 
         public void SendSmsInBackground(string recipient, string message = null)
         {
-            message = message ?? string.Empty;
+            if (!CanSendSmsInBackground)
+            {
+                OnSmsDeliveryResult?.Invoke(isSuccessful: false);
+
+                return;
+            }
+
+            message ??= string.Empty;
+
+            _smsManager = SmsManager.Default;
 
-            if (CanSendSmsInBackground)
+            var piSent = PendingIntent.GetBroadcast(CONTEXT, 0, new Intent(SMS_SENT_INTENT_ACTION), 0);
+            var piDelivered = PendingIntent.GetBroadcast(CONTEXT, 0, new Intent(SMS_DELIVERY_INTENT_ACTION), 0);
+
+            CONTEXT.RegisterReceiver(this, new IntentFilter(SMS_SENT_INTENT_ACTION));
+            CONTEXT.RegisterReceiver(this, new IntentFilter(SMS_DELIVERY_INTENT_ACTION));
+
+            _smsManager.SendTextMessage(recipient, null, message, piSent, piDelivered);
```

Comment:
Why did you change send method? in prevoius we use SendMultipartTextMessage

Reply:
**TD;DL:**
For Resa project this this new method is fine and is OK. But for other people we need to use previous method. But it would be a bit complex and time-consuming to implement that.

Because the result mechanism works only for one SMS. The previous method i.e SendMultipartTextMessage will accept message bodies in various lengths (even longer than one SMS limit) and then split them to be fit in one SMS limit. But in that case it would be more co,plex to me to support sending result(s) of multiple SMS. Because every single SMS has its own BroadcastReceiver event and combining all receiver events into one single boolean result would take more time to implement and it is a bit complex. The new send method is perfectly OK for RESA project, but for other people using this library I certainly need to use previous method and support getting result for multiple SMSs.

Reply:
Please bring back.

Reply:
I did bring it back.